index.js

```diff
@@ -0,0 +1,98 @@
+const express = require('express');
+const cors = require('cors');
+const bcrypt = require('bcryptjs')
+const session = require('express-session')
+// require your store library
+const KnexSessionStore = require('connect-session-knex')(session);
+
+const db = require('./database/dbConfig.js');
+
+const server = express();
+
+server.use(express.json());
+server.use(cors());
+
+// configure express-session middleware
+const sessionConfig = {
+      name: 'notsession', // default is connect.sid
+      secret: 'nobody tosses a dwarf!',
+      cookie: {
+        maxAge: 1 * 24 * 60 * 60 * 1000,
+        secure: false, // only set cookies over https. Server will not send back a cookie over http.
+      }, // 1 day in milliseconds
+      httpOnly: true, // don't let JS code access cookies. Browser extensions run JS code on your browser!
+      resave: false,
+      saveUninitialized: false,
+      store: new KnexSessionStore({
+        tablename: 'sessions',
+        sidfieldname: 'sid',
+        knex: db,
+        createtable: true,
+        clearInterval: 1000 * 60 * 60,
+      }),
+    }
+
+server.use(session(sessionConfig));
+
+// testing cookies
+server.get('/setname', (req, res) => {
+    req.session.name = 'Frodo';
+    res.send('got it');
+  });
+  
+server.get('/greet', (req, res) => {
+    const name = req.session.name;
+    res.send(`hello ${req.session.name}`);
+  });
+// close cookie test
+
+server.get('/', (req, res) => {
+  res.send('Server is humming along nicely.');
+});
+
+// protect this route, only authenticated users should see it
+server.get('/api/users', (req, res) => {
+  db('users')
+    .select('id', 'username', 'password')
+    .then(users => {
+      res.json(users);
+    })
+    .catch(err => res.send(err));
+});
+
+// login
+server.post('/api/login', (req, res) => {
+    // grad creds
+    const creds = req.body;
+    // find the user
+    db('users').where({username: creds.username}).first().then(user => {
+    // check creds
+    if (user && bcrypt.compareSync(creds.password, user.password)) {
+        req.session.username = user.username;
+        res.status(200).send(`Welcome ${req.session.username}`);
+    } else {
+        res.status(401).json({ message: 'You shall not pass!'});
+    }
+    }).catch(err => res.status(500).send(err))
```

Comment:
I'd recommend console.logging the error in your `catch()` block. The error tends to be shortened in PostMan, and conceptually, it's not the client's job to fix your server, so they don't really need to know the exact error.

```js
// snippet
.catch(err => {
    console.log('/api/loginPOST error:', err);
    res.status(500).send('Please try again later.');
});
```